File: iosApp/deadly/Core/Model/ShowArtworkService.swift
import Foundation
import Observation

@Observable
final class ShowArtworkService: @unchecked Sendable {
    static let shared = ShowArtworkService()
    @ObservationIgnored private var sourceTypes: [String: RecordingSourceType] = [:]
    @ObservationIgnored private let lock = NSLock()
    var badgeStyle: SourceBadgeStyle = .long

    func sourceType(for recordingId: String) -> RecordingSourceType? {
        lock.lock()
        defer { lock.unlock() }
        return sourceTypes[recordingId]
    }

    func populate(_ entries: [String: RecordingSourceType]) {
        lock.lock()
        defer { lock.unlock() }
        sourceTypes.merge(entries) { _, new in new }
    }
}

File: androidApp/core/model/src/main/java/com/grateful/deadly/core/model/ShowArtworkService.kt
package com.grateful.deadly.core.model

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import java.util.concurrent.ConcurrentHashMap

object ShowArtworkService {
    private val sourceTypes = ConcurrentHashMap<String, RecordingSourceType>()

    var badgeStyle: SourceBadgeStyle by mutableStateOf(SourceBadgeStyle.LONG)

    fun sourceType(recordingId: String): RecordingSourceType? = sourceTypes[recordingId]

    fun populate(entries: Map<String, RecordingSourceType>) {
        sourceTypes.putAll(entries)
    }

    fun clear() {
        sourceTypes.clear()
    }
}
